src/insights_client/metrics.py

```diff
@@ -56,6 +83,13 @@ def __init__(
         cfg = configparser.RawConfigParser()
         cfg.read(config_file)
 
+        # TODO: switch to this once the core PR is merged
+        # self.offline = InsightsConfig().load_all().offline
+        self.offline = _is_offline(cfg)
+        if self.offline:
+            logger.debug("Metrics: Offline mode. No metrics will be sent.")
+            return
+
```

Comment:
It this the place where we want to do it? I can imagine extracting the decision to a function and then using a dummy class instead. But that would require extract the configuration file loading too, which may not be that bad after all.

```python
def _load_configs(config_file="/etc/insights-client/insights-client.conf", rhsm_config_file="/etc/rhsm/rhsm.conf")
    cfg = configparser.RawConfigParser()
    cfg.read(config_file)

    rhsm_cfg = configparser.ConfigParser()
    rhsm_cfg.read(rhsm_config_file)
    
    return cfg, rhsm_cfg

def get_metrics_client():
    cfg, rhsm_cfg = _load_configs()
    if _is_offline(cfg):
        return OfflineMetricsClient()
    else
        return HTTPMetricsClient(config=cfg, rhsm_config=rhsm_cfg)

class OfflineMetricsClient:
    def __init__():
        logger.debug("Metrics: Offline mode. No metrics will be sent.")

    def post(self, event):
        pass
```

Yes, I see that it would result in a RHSM config file being loaded and parsed even in the offline mode.

But as I see it further, there are more cases when the function falls back to being offline. That makes me think more about extracting the configuration. We probably don’t want to do it here in this pull request.